### ISSUE TITLE
megaraid_sas module not working on Dell R330 server

### ISSUE THREAD

**Issue by taxilian** [coreos/bugs#970]
Standard Dell R330 server configuration server, the RAID controller doesn't load and thus drives on it can't be used.  I'm using the alpha branch of CoreOS.  The block devices simply don't show up as being detected, and bunches of errors show up in the dmesg (so many that it overflows during boot and you don't see the beginning).
[lspci.txt](https://github.com/coreos/bugs/files/31450/lspci.txt)

I found a similar-looking issue here: https://centoros.wordpress.com/2015/07/22/iommu-dmar-fault/
The suggested fix to add a kernel parameter `intel_iommu=pt` did not work.  The same configuration works if I boot the centos7 installer.

Kernel version and dmesg output from coreos and centos7 are attached
[dmesg_centos7.txt](https://github.com/coreos/bugs/files/31440/dmesg_centos7.txt)
[dmesg_coreos.txt](https://github.com/coreos/bugs/files/31441/dmesg_coreos.txt)

lspci describes my card as:

    LSI Logic / Symbios Logic MegaRAID SAS 2008 [Falcon] (rev 03)

**Comment by taxilian** [comment]
Quick update, this works fine if we disable virtualization on the motherboard; are there any downsides to doing this?

**Comment by crawford** [comment]
There shouldn't be any downsides to disabling VT-d. None of the software on CoreOS uses virtualization by default.

**Comment by mischief** [comment]
@taxilian we will be updating to linux 4.3.0 in the next alpha. it's a long shot, but might be worth trying again.

**Comment by taxilian** [comment]
Understood.  Unfortunately the servers in question are all in production now, so I'm not going to be able to verify it for you; as an additional data point, I experienced an identical-looking issue on a supermicro barebones server that is a couple of years old which was using an after-market LSI PCI RAID card (I can't find the exact model); disabling virtualization did not help on that server, and I ended up having to remove the RAID card, since that machine it wasn't mandatory.

I'll let you know if I find anything else, but all my servers are reinstalled for now so I don't have current plans to install on anything new.
